### PR TITLE
deepsight:complete && deepsight:incomplete

### PR DESCRIPTION
**Pull request by delphiactual**
Add ~~`is:deepsightcomplete` filter~~ `deepsight` keyword

### CODE REVIEW

File: src/app/search/search-filters/sockets.tsx [hunk 1]
```diff
@@ -180,24 +180,46 @@ const socketFilters: FilterDefinition[] = [
...
-              // but not the objectiveless stub used to blank out that socket
-              plugDef.objectives
-            );
-          })
-      ),
+    suggestions: ['complete', 'incomplete'],
+    filter:
+      ({ filterValue }) =>
+      (item: DimItem) => {
+        if (filterValue === 'deepsight') {
```
Comment (bhollis) [inline src/app/search/search-filters/sockets.tsx:189]: `switch`?

File: src/app/search/search-filters/sockets.tsx [hunk 2]
```diff
@@ -180,24 +180,45 @@ const socketFilters: FilterDefinition[] = [
...
+          case 'incomplete':
+            return isDeepsight(item, false);
+        }
+      },
   },
 ];
 
 export default socketFilters;
+
+function isDeepsight(item: DimItem, checkComplete: Boolean | null = null) {
```
Comment (nev-r) [inline src/app/search/search-filters/sockets.tsx:203]:
```suggestion
function isDeepsight(item: DimItem, checkComplete?: Boolean) {
```

File: src/app/search/search-filters/sockets.tsx [hunk 3]
```diff
@@ -180,24 +180,45 @@ const socketFilters: FilterDefinition[] = [
...
+  return Boolean(
+    item.bucket.inWeapons &&
+      item.sockets?.allSockets.find((s) => {
+        const plugDef = s.plugged?.plugDef;
+        const complete = // always true for null, checkcomplete XNOR complete
+          checkComplete === null
```
Comment (nev-r) [inline src/app/search/search-filters/sockets.tsx:209]:
```suggestion
          checkComplete === undefined
```
could skip the entire default null assignment above by just checking for the lack of a param here

File: src/app/search/search-filters/sockets.tsx [hunk 4]
```diff
@@ -180,24 +180,45 @@ const socketFilters: FilterDefinition[] = [
...
+      },
   },
 ];
 
 export default socketFilters;
+
+function isDeepsight(item: DimItem, checkComplete: Boolean | null = null) {
+  return Boolean(
+    item.bucket.inWeapons &&
+      item.sockets?.allSockets.find((s) => {
```
Comment (nev-r) [inline src/app/search/search-filters/sockets.tsx:206]: should the completeness check be calculated for every single socket during the .find?
might make more sense to just .find the deepsight socket, then afterward (and if it even exists) check its completeness

**Review by nev-r — APPROVED** [review]

**Pull request merged**